hack/generator/cmd/gen/gen.go

```diff
@@ -15,13 +16,30 @@ import (
 	"k8s.io/klog/v2"
 )
 
+type cancellableTransport struct {
+	ctx   context.Context
+	inner http.RoundTripper
+}
+
+func (transport *cancellableTransport) RoundTrip(request *http.Request) (*http.Response, error) {
```

Comment:
Is this impl-ing an interface? Should we have an assertion that the interface we want is impled?